pom.xml

```diff
@@ -846,13 +896,331 @@
                 <plugin>
                     <groupId>org.codehaus.mojo</groupId>
                     <artifactId>build-helper-maven-plugin</artifactId>
-                    <version>3.2.0</version>
+                    <version>3.3.0</version>
                 </plugin>
                 <plugin>
                   <groupId>org.apache.maven.plugins</groupId>
                   <artifactId>maven-antrun-plugin</artifactId>
-                  <version>3.0.0</version>
+                  <version>3.1.0</version>
                   <executions>
+                    <execution>
```

Comment:
Seems like there's conceptually two unrelated changes here:
- moving the create-source-path-properties stuff up to the parent pom
- adding the optional shimplify task

Would be nice to see these as separate PRs, as it doesn't seem like the create-source-path-properties change requires shimplify.  It has merit on its own and is more than half the code change in this PR.